### PR TITLE
Add qi overview

### PR DESCRIPTION
**Pull request by yte0**

### CODE REVIEW

**Comment by arnfinn** [comment]
@yte0: det fungerte ikke helt som jeg forventet.

Knappen *Alle* er inaktiv. Jeg hadde forventet at filtrering nulles ved å trykke på denne.

Filtervalg akkumuleres. Så hvis jeg trykker *Hjerte* vil `clicked_register` være 
```
[1] "ablanor" "noric" 
```
men hvis jeg så trykker på *Kreft* blir `clicked_register`
```
[1] "ablanor"    "noric"      "barnekreft" "brystkreft" "gynkreft"   "lungekreft" "lymfoid"    "melanom"    "prostata"   "tarmkreft" 
```
Jeg hadde forventet
```
[1] "barnekreft" "brystkreft" "gynkreft"   "lungekreft" "lymfoid"    "melanom"    "prostata"   "tarmkreft" 
```

Er det enkelt å fikse dette?

Ellers ser det veldig bra ut:tada:

**Comment by yte0** [comment]
skal ikke være vanskelig å gjøre de endringene. i'm on it, boss :-)

File: R/qi_overview.R [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+        )
+      )
+    ),
+    purrr::pmap(
+      list(
+        category_name,
+        id,
+        nr_of_reg
+      ),
+      function(x, y, z) {
```
Comment (arnfinn) [inline R/qi_overview.R]: Can `x`, `y` and `z` have meaningful names?

File: R/table_functions.R [hunk 2]
```diff
@@ -78,20 +78,20 @@ group_data <- function(data_list, by) {
     data_list[["indicator"]] <- add_orgnr(data_list, by) %>%
       dplyr::group_by(
         .data[[org_column]],
-        .data[[config[["data"]][["column"]][["year"]]]],
+        .data[[config$data$column$year]],
```
Comment (arnfinn) [inline R/table_functions.R]: as above

File: R/mod_quality_overview.R [hunk 3]
```diff
@@ -204,4 +223,49 @@ mod_quality_overview_server <- function(input,
...
+       )
+      }
+    )
+    shiny::observeEvent(
+      input$alle, {
+        filter_indicator$indicator <- register_data[["description"]] %>%
+        purrr::pluck("IndID") %>%
+        unique()
+      })
+   })
```
Comment (arnfinn) [inline R/mod_quality_overview.R]: These lines are not tested

File: R/table_functions.R [hunk 4]
```diff
@@ -52,13 +52,13 @@ add_orgnr <- function(data_list, by = "RHF") {
...
+            .data[[config$data$column$unit_id$sh]],
+            .data[[config$data$column$unit_name$sh]]) %>%
           dplyr::mutate(
-            "OrgNrShus" = as.character(.data[[config[["data"]][["column"]][["unit_id"]][["sh"]]]])),
-        by = config[["data"]][["column"]][["unit_id"]][["sh"]]
+            "OrgNrShus" = as.character(.data[[config$data$column$unit_id$sh]])),
+        by = config$data$column$unit_id$sh
       ) %>%
-      dplyr:: filter(!is.na(.data[[config[["data"]][["column"]][["unit_name"]][["sh"]]]]))
+      dplyr:: filter(!is.na(.data[[config$data$column$unit_name$sh]]))
```
Comment (arnfinn) [inline R/table_functions.R]: Are these changes related to current PR? Maybe we should make a `config` PR where we fix all the `config` stuff?

File: R/table_functions.R (hunk 2 above)
Comment (arnfinn) [reply R/table_functions.R]: or below... config stuff

**Comment by arnfinn** [comment]
One more thing... When I click on "fagområde" not in the current data, the filter is reset to "all" (`filter_indicator$indicator` is set to `NULL`?). Is it not better to filter on what you click and just remove everything?

File: R/table_functions.R (hunk 4 above)
Comment (arnfinn) [reply R/table_functions.R]: BTW, not a big deal... I am fine with merging the PR including these changes.

File: R/qi_overview.R (hunk 1 above)
Comment (yte0) [reply R/qi_overview.R]: ok, will get that fixed:-)

Comment (arnfinn) [reply R/qi_overview.R]: If it is coordinates it is perfect as is:)

File: R/table_functions.R (hunk 4 above)
Comment (yte0) [reply R/table_functions.R]: I thought that since there is nothing major there, I would add it here. but I agree that it could have been gathered in a separate PR.

**Comment by yte0** [comment]
That is right. It does so for the yml entries with registery names in the description table. However, if the registery names in the yml file are not found in the discription table, we cant filter based on the indicators. Since the indicators are in the description table.  As several of the registers get listed on the description table, the filter will work as intended.

File: R/mod_quality_overview.R [hunk 5]
```diff
@@ -100,8 +100,7 @@ mod_quality_overview_server <- function(input,
       register_data[["hospital_name_structure"]] %>%
           dplyr::select(
             .data[[config$data$column$unit_name$hfshort]],
-            .data[[config$data$column$unit_id$hf]]
-
+            .data[[config$data$column$unit_id$hf]],
```
Comment (arnfinn) [inline R/mod_quality_overview.R]: OK with comma here?

File: R/mod_quality_overview.R (hunk 3 above)
Comment (arnfinn) [reply R/mod_quality_overview.R]: Is it possible to make a test?

File: R/mod_quality_overview.R (hunk 5 above)
Comment (yte0) [reply R/mod_quality_overview.R]: did not notice that one. I will remove it.  you are getting very good at the reviewing:-)

**Comment by yte0** [comment]
not sure how i should do the testing, but I will try to find out.

File: R/mod_quality_overview.R (hunk 5 above)
Comment (arnfinn) [reply R/mod_quality_overview.R]: Close to perfect:)

**Comment by arnfinn** [comment]
We can implement testing later, if it is too complicated at the moment:)

File: R/mod_quality_overview.R (hunk 3 above)
Comment (arnfinn) [reply R/mod_quality_overview.R]: We can do it later

**Comment by arnfinn** [comment]
Great job!

**Comment by arnfinn** [comment]
PUSH THE BUTTON!

**Comment by yte0** [comment]
which button:)

**Comment by arnfinn** [comment]
Squash and merge...

17. mar. 2020 kl. 16:25 skrev YT <notifications@github.com>:

﻿

which button:)
